pykechain/models/part2.py

```diff
@@ -228,13 +228,28 @@ def populate_descendants(self, batch: int = 200) -> Union['Partset', List['Part2
         >>> bike.populate_descendants(batch=150)
 
         """
-        descendants_flat_list = list(self._client.parts(
-            parent_id=self.id,
+        all_descendants = list(self._client.parts(
             category=self.category,
-            batch=batch)
-        )
+            batch=batch,
+            descendants=self.id,
+        ))[1:]  # remove the part itself, which is returned on index 0
+
+        # Create mapping table from a parent part ID to its children
+        children_by_parent_id = dict()
+        for descendant in all_descendants:
+            if descendant.parent_id in children_by_parent_id:
+                children_by_parent_id[descendant.parent_id].append(descendant)
+            else:
+                children_by_parent_id[descendant.parent_id] = [descendant]
+
+        # Populate every descendant with its children
+        for descendant in all_descendants:
+            if descendant.id in children_by_parent_id:
+                descendant._cached_children = children_by_parent_id[descendant.id]
+
+        self._cached_children = children_by_parent_id.get(self.id, list())
 
-        self._cached_children = descendants_flat_list
+        return all_descendants
```

Comment:
not cached children return? I do not get this why this would be faster and less traffic. It does not seem to make use of in-object caching / memoization